src/firefly/java/edu/caltech/ipac/table/io/FITSTableReader.java

```diff
@@ -393,9 +556,117 @@ private static DataGroup convertStarTableToDataGroup(StarTable table,
         }
         dataGroup.trimToSize();
         return dataGroup;
+    }*/
+
+
+    //This function is based on the packageValue function from the FitsStarTable class in the uk.ac.starlink.fits package
+    private static Object getValAsObject(Object base, int icol, boolean[] hasBlank, long[] blanks, boolean[] isScaled,
+                                         double[] scales, double[] zeros) {
+        if (base == null) {
+            return null;
+        } else {
+            Class bcls = base.getClass();
+            Class cls = bcls.isArray() ? bcls.getComponentType() : null;
+            if (cls != null && Array.getLength(base) == 1) {
+                boolean hasblank = hasBlank[icol];
+                long blank = blanks[icol];
+                boolean scaled = isScaled[icol];
+                double scale = scales[icol];
+                double zero = zeros[icol];
+                if (cls == Byte.TYPE) {
+                    byte val = ((byte[])(base))[0];
+                    /*
+                    TODO: Attempt to use ternary operator here instead
+                     */
+                    if (hasblank && val == (byte)((int)blank))
```

Comment:
you don't need to cast blank to do comparisons. Also since this is so repeated you might want to put it on one line.
```java
if (hasblank && val == blank) return null;
```